lib/core.js

```diff
@@ -35,12 +35,12 @@ function Promise(fn) {
       deferreds.push(deferred)
       return
     }
+    var cb = state ? deferred.onFulfilled : deferred.onRejected
+    if (cb === null) {
+      (state ? deferred.resolve : deferred.reject)(value)
+      return
+    }
     asap(function() {
-      var cb = state ? deferred.onFulfilled : deferred.onRejected
-      if (cb === null) {
-        (state ? deferred.resolve : deferred.reject)(value)
-        return
-      }
```

Comment:
Having ordering issues in code where substituting these two equivalent cases for each other changes the issue is a big no